src/Compilers/CSharp/Test/Syntax/Parsing/DeclarationScopeParsingTests.cs

```diff
@@ -3662,5 +3950,614 @@ public void Using_04_RefScoped(LanguageVersion langVersion)
+                                N(SyntaxKind.RefType);
+                                {
+                                    N(SyntaxKind.RefKeyword);
+                                    N(SyntaxKind.PredefinedType);
+                                    {
+                                        N(SyntaxKind.IntKeyword);
+                                    }
+                                }
+                            }
+                            N(SyntaxKind.VariableDeclarator);
+                            {
+                                N(SyntaxKind.IdentifierToken, "F3");
+                            }
+                        }
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(LanguageVersion.CSharp10)]
+        [InlineData(LanguageVersion.CSharp11)]
+        public void Field_02(LanguageVersion langVersion)
+        {
+            string source =
+@"
+ref struct R2
+{
+    const scoped int F3;
+}
+";
+            UsingTree(source, TestOptions.Regular.WithLanguageVersion(langVersion),
+                // (4,18): error CS1001: Identifier expected
+                //     const scoped int F3;
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "int").WithLocation(4, 18),
+                // (4,18): error CS0145: A const field requires a value to be provided
+                //     const scoped int F3;
+                Diagnostic(ErrorCode.ERR_ConstValueRequired, "int").WithLocation(4, 18),
+                // (4,18): error CS1002: ; expected
+                //     const scoped int F3;
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "int").WithLocation(4, 18)
+                );
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.StructDeclaration);
+                {
+                    N(SyntaxKind.RefKeyword);
+                    N(SyntaxKind.StructKeyword);
+                    N(SyntaxKind.IdentifierToken, "R2");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.FieldDeclaration);
+                    {
+                        N(SyntaxKind.ConstKeyword);
+                        N(SyntaxKind.VariableDeclaration);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "scoped");
+                            }
+                            M(SyntaxKind.VariableDeclarator);
+                            {
+                                M(SyntaxKind.IdentifierToken);
+                            }
+                        }
+                        M(SyntaxKind.SemicolonToken);
+                    }
+                    N(SyntaxKind.FieldDeclaration);
```

Comment:
>N(SyntaxKind.FieldDeclaration); [](http://example.com/codeflow?start=20&length=31)

Should this be a single field with a `ScopedTypeSyntax` instead?

Reply:
> Should this be a single field with a ```ScopedTypeSyntax instead```?

I guess we could represent it this way.

Reply:
>>Should this be a single field with a ScopedTypeSyntax instead?

>I guess we could represent it this way.

Actually, now I think that would be incorrect. Here is the grammar for constant declaration:
```
constant_declaration
    : attributes? constant_modifier* 'const' type constant_declarators ';'
    ;
```

The way I read it, immediately after ```const```, "scoped" is not a keyword. It should be treated as an identifier. And we do, I didn't change the logic around determining if "scoped" is a keyword. Bottom line, ```ScopedTypeSyntax``` shouldn't be used to represent constant field declaration.